warehouse/bigquery.go

```diff
@@ -102,19 +101,16 @@ func (bq *BigQuery) LastSyncPoint(_ context.Context) (time.Time, error) {
 	return t, nil
 }
 
-func (bq *BigQuery) SaveSyncPoints(_ context.Context, bundles ...client.ExportMeta) error {
+func (bq *BigQuery) SaveSyncPoint(_ context.Context, endTime time.Time) error {
 	if err := bq.connectToBQ(); err != nil {
 		return err
 	}
 	defer bq.bqClient.Close()
 
-	values := make([]string, len(bundles))
-	for i, p := range bundles {
-		values[i] = fmt.Sprintf("(%d, TIMESTAMP(\"%s\"), TIMESTAMP(\"%s\"))", p.ID, time.Now().UTC().Format(time.RFC3339), p.Stop.UTC().Format(time.RFC3339))
-	}
+	value := fmt.Sprintf("(%d, TIMESTAMP(\"%s\"), TIMESTAMP(\"%s\"))", -1, time.Now().UTC().Format(time.RFC3339), endTime.UTC().Format(time.RFC3339))
```

Comment:
What is -1 going to give you for ID? Is that going to be problematic?

Reply:
Also only inserting one record is not going to be an issue with the `removeSyncPointsAfter` behavior above? I guess the `bundles` are always day aligned so I guess it should be okay?

Reply:
> What is -1 going to give you for ID? Is that going to be problematic?

We don't use the IDs that are stored in the database table. We just query for the max timestamp in the last column (bundle end time). I thought about synthesizing the IDs, but I don't think that really buys us anything in with the new API paradigm.

> Also only inserting one record is not going to be an issue with the removeSyncPointsAfter behavior above? I guess the bundles are always day aligned so I guess it should be okay?

I don't think this should be an issue since `removeSyncPointsAfter` is just making sure the the last BundleEndTime is greater than the last `EventStart`.

However, this does highlight an underlying assumption that the bundle time ranges are always day aligned. We'll need to be careful about this as we move to the new API. Maybe we just keep enforcing that the time ranges/durations are day aligned.

Reply:
If we are always using `-1` though, do we run into problem when checkpoints are deleted that aren't day aligned? If there is only ever 1 entry, won't that trigger us to start back from the beginning rather than just start of day?

Reply:
> If there is only ever 1 entry

I'm pretty sure that it's just inserting rows, not overwriting them, so there will be a row for each sync.

Reply:
That's what I'm asking though. `-1` will give a unique ID automatically? Or is `ID` not even unique at all and all of these warehouses use some other autogenned identifier? I think that's the part I am missing.

Reply:
`ID` isn't a special field, it's just an integer. And it doesn't look like bigquery or redshift treat them special.

Reply:
Ah, okay, that was the part I was missing!